doc/developers-guide/accumulators.md

```diff
@@ -30,132 +30,111 @@ a) performance - if we need to do something involving inspecting a stanza or mor
 
-Some attributes are calculated upon request and stored: these are `command`, `xmlns` and `iq_query_info`.
-If they are needed, you should first use `mongoose_acc:require/2` and then just get:
+A constructor for accumulators. `new_acc_params()` is a map with following supported keys:
 
-```
-Acc1 = mongoose_acc:require(command, Acc),
-Cmd = mongoose_acc:get(command, Acc1)
-```
+* `location` - Should be a `{Module, Function, Line}` tuple (may be constructed with `?LOCATION` macro from `mongoose.hrl`). Its format is not enforced by the acc logic but Dialyzer will most probably complain about any other type.
+* `lserver` - Nameprepped domain, which is a context for current processing.
+* `element` (optional) - If present, will be used as a source for `stanza` map.
+* `from_jid`, `to_jid` (optional) - Values used to override `from` and `to` attributes of the `element`, respectively.
 
-You can set whatever attributes you want, but treat them as read-only - an accumulator is meant
-to accumulate results and carry them forward with it, not to introduced a semi-mutable data structure.
-If you put a value to a key which already exists it will work, at least as of today, but may also produce a warning.
-Some day it may stop working, you have been warned.
+If `element` is provided, the call will fail with an exception, if it's impossible to extract both sender and recipient JID, either from stanza or `to_jid` and `from_jid`.
 
-If you need to store multiple values under the same key, then either:
-* use mongoose_acc:append/3
-* or generate a unique derivative key for each use
+While allowed, it should be avoided to create stanza-less accumulators.
 
-There is one exception: if you really must treat an acc to carry just one value back from a call,
-use a key `result`, this is not monitored.
+### Getters for predefined fields
 
-For mass updates (multiple key at once) consider using `mongoose_acc:update/2`, it may be
-slightly more efficient. Be aware, however, that it will not warn you if you
-overwrite an existing key.
+* `ref(t())`
+* `timestamp(t())`
+* `lserver(t())`
+* `element(t())`
+* `stanza_name(t())` - Returns `name` value from `stanza` map.
+* `stanza_type(t())` - Returns `type` value from `stanza` map.
+* `stanza_ref(t())` - Returns `ref` value from `stanza` map. This is not the same as `ref(t())`!
 
-## Stripping and "persistence"
+### `update_stanza(stanza_params(), t())`
 
-Accumulator is used mostly to cache values for reuse within a c2s process; when it goes out to somewhere else it is stripped of all its attrributes except some used for tracing (ref, timestamp).
-If you want it to carry some values along with it use a dedicated api for setting "persistent" properties:
-```
-Acc2 = mongoose_acc:add_prop(myprop, 123, Acc1),
-V = mongoose_acc:get_prop(myprop, Acc2).
-```
+Replaces whole `stanza` field in accumulator with params provided in `stanza_params()`, which is a map of 3 fields: `element`, `from_jid`, `to_jid`.
```

Comment:
Replaces the whole `stanza` field in accumulator with params provided in `stanza_params()`, which is a map of 3 fields: `element`, `from_jid`, `to_jid`.